e2e/test/scenarios/native/reproductions/16584-run-selected-text-does-not-preserve-parameter-values.cy.spec.ts

```diff
@@ -0,0 +1,24 @@
+import { restore, openNativeEditor } from "e2e/support/helpers";
+
+describe("issue 16584", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsNormalUser();
+    cy.intercept("POST", "/api/dataset").as("run");
+  });
+
+  it("should pass parameters when running with 'Run select text' (metabase#16584)", async () => {
+    const editor = openNativeEditor();
+
+    editor.type("SELECT * FROM ACCOUNTS WHERE COUNTRY = {{ country ");
+    editor.type("{selectAll}");
```

Comment:
Please don't do this.
If you take a look at the helper `openNativeEditor()`, you'll see that it actually visits a home page and then, starts the new > native query UI flow all over again. Apart from it being wasteful, it will not even work in this case.

Btw, please try to avoid assigning selectors to a `const` and use them directly.

Reply:
I've fixed the assigning to a `const` bit, but I'm unclear as to why I shouldn't be using `openNativeEditor`, as it is currently being using in at least 48 test suites? Is there an alternative I should be using instead?

As to "it will not even work in this case": the test reproduces the issue and shows that the fix fixes it, can you elaborate on why you think it would not work?

Reply:
> Apart from it being wasteful

We need at least one `cy.visit` call somewhere. Am I missing something?

> it will not even work in this case.

Can you elaborate on what will not work?

> Btw, please try to avoid assigning selectors to a `const` and use them directly.

Why?

Reply:
> Why?

https://learn.cypress.io/cypress-fundamentals/understanding-the-asynchronous-nature-of-cypress

https://docs.cypress.io/guides/core-concepts/variables-and-aliases#Variables